### PR TITLE
Change helm scripting

### LINKED ISSUE #213: [Bug Report]: aks-run-helm doesn't work
### Module path

/modules/deployment-scripts/aks-run-helm

### Bicep version

Bicep CLI version 0.12.40 (41892bd0fb)

### Describe the bug

`helm upgrade --install ${app.helmApp} ${app.helmAppName}` will not work in combination with `name: 'helmInstall-${app.helmAppName}-${I}'`:

To install `cert-manager` for instance, in CLI you would run `helm upgrade --install cert-manager jetstack/cert-manager`. If that relates to below code it runs `helm upgrade --install jetstack/cert-manager cert-manager` instead, which fails silently (related to #174)
```bicep
helmRepo: 'jetstack'
helmRepoURL: 'https://charts.jetstack.io'
helmApps: [
  {
    helmAppName: 'cert-manager'
    helmApp: 'jetstack/cert-manager'
    helmParams: '--version v1.10.1 --namespace cert-manager --create-namespace --set installCRDs=true'
  }
]
```

If we change `helmAppName` and `helmApp`, `name: 'helmInstall-${app.helmAppName}-${I}'` contains an `/`, resulting in
```
The template resource 'helmInstall-jetstack/cert-manager-0' for type 'Microsoft.Resources/deployments' at line '1' and column '3453' has incorrect segment lengths. A nested resource type must have identical number of segments as its resource name. A root resource type must have segment length one greater than its resource name. Please see https://aka.ms/arm-template/#resources for usage details.'.", "additionalInfo": [{"type": "TemplateViolation", "info": {"lineNumber": 1, "linePosition": 3453, "path": "properties.template.resources[0].type"}}]}
```

### To reproduce

```
  name: 'kubectlGetNodes'
  params: {
    useExistingManagedIdentity: true
    managedIdentityName: aksrunid.name
    aksName: aks.name
    location: location
    helmRepo: 'jetstack'
    helmRepoURL: 'https://charts.jetstack.io'
    helmApps: [
      {
        helmAppName: 'jetstack/cert-manager'
        helmApp: 'cert-manager'
        helmParams: '--version v1.10.1 --namespace cert-manager --create-namespace --set installCRDs=true'
      }
    ]
  }
```

### Code snippet

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by aristosvo**
## Description

The current helm scripting is not working. That has 2 reasons:
- Multiline (every line is executed in a different container which don't share state)
- the order of the arguments given to helm. An example of that here:
[helm update --install cert-manager jetstack/cert-manager](https://cert-manager.io/docs/installation/helm/#4-install-cert-manager) == helm update --install `helmAppName` `helmApp`

Closes #213, #231

## Updating an existing module

- [x] This is a bug fix:
  - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](../CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [x] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.

### CODE REVIEW

**Comment by dciborow** [comment]
@aristosvo , 

Looks good (and looks like I switched the params by mistake). Minor update to also fix the name in the test.
https://github.com/aristosvo/bicep-registry-modules/pull/1

**Review by dciborow — CHANGES_REQUESTED** [review]
Please update unit test as well.

**Comment by aristosvo** [comment]
@dciborow I also included in my own setup an extra output which makes validation of the output possible. Should I add that as well and is there a possibility to check the exitCode in the unit test?

```bicep
output helmOutputs array = [for (app, i) in helmApps: {
  appName: app.helmAppName
  outputs: helmAppInstalls[i].outputs.commandOutput
}]
```

File: modules/deployment-scripts/aks-run-helm/main.bicep [hunk 1]
```diff
@@ -53,3 +52,8 @@ module helmAppInstalls 'br/public:deployment-scripts/aks-run-command:1.0.1' = [f
     cleanupPreference: cleanupPreference
   }
 }]
+
+output helmOutputs array = [for (app, i) in helmApps: {
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/main.bicep:56]: @description('Helm Command Output Values')

Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/main.bicep:56]:
```suggestion
@description('Helm Command Output Values')
output helmOutputs array = [for (app, i) in helmApps: {
```

Comment (dciborow) [reply modules/deployment-scripts/aks-run-helm/main.bicep:56]: @aristosvo , we are so close. Just need to add this description field to the new output.

**Review by dciborow — APPROVED** [review]

File: modules/deployment-scripts/aks-run-helm/main.bicep (hunk 1 above)
Comment (dciborow) [reply modules/deployment-scripts/aks-run-helm/main.bicep:56]: @shenglol , these changes are all set once all the checks pass. Looks like we will need you to approve as well.

**Comment by shenglol** [comment]
@aristosvo Could you update one module at a time? Changing multiple modules in one PR is not supported.

**Comment by aristosvo** [comment]
> @aristosvo Could you update one module at a time? Changing multiple modules in one PR is not supported.

Sure. I’m currently not close to a pc, but I reverted most of the changes from the GH UI. Feel free to fix the new line change if that is really necessary.

**Comment by Gordonby** [comment]
@aristosvo - One of the PR checks is failing. Can you `brm generate` and push to fix?

**Comment by aristosvo** [comment]
@Gordonby Sure, I've tried to fix it, but this wasn't a change to what already was there previously :)

**Review by shenglol — APPROVED** [review]

**Pull request merged**